src/hardware/sblaster.cpp

```diff
@@ -620,15 +620,14 @@ static void PlayDMATransfer(Bitu size)
 /*
  *  This function intercepts the first DMA transfer and decides if it should
  *  suppress or play it.  The criteria is as follows:
- *   - Suppress if the card is not in a DMA mode (ie: junk data)
- *   - Suppress DWORD-sized transfers (or less) in 8-bit DMA modes
+ *   - Suppress DWORD-sized transfers (or less) in non-16-bit DMA modes
  *   - Suppress BYTE-sized transfers in 16-bit DMA modes
  */
 static void SuppressInitialDMATransfer(Bitu size)
 {
-	if (sb.dma.mode == DSP_DMA_NONE ||
-	    (sb.dma.mode < DSP_DMA_16 && size <= sizeof(uint32_t)) ||
-	    size <= sizeof(uint8_t)) {
+	const uint8_t size_limit = sb.dma.mode < DSP_DMA_16 ? sizeof(uint32_t)
+	                                                    : sizeof(uint8_t);
+	if (size <= size_limit) {
```

Comment:
Nice, it's more readable now :) So for DSP_DMA_NONE we want to proceed to PlayDMATransfer anyway? That's… suprising a bit, when looking from the outside, but if it works (I haven't tested it myself).

As for the change itself: I would argue for using `size_t` in line 628:
```
const size_t size_limit = …
```
… but it does not really matter - I'll let you decide. 

It does not really matter because even -O1 optimization will turn this code as if size_t was used; with -O0 slightly different assembly is being generated, but there's no performance difference: https://godbolt.org/z/CV2BS-

(with -O0 GCC generates movzx for uint8_t and mov for size_t; as far as I can tell it's the only difference; the same number of instructions and the same number of registers is being used)

Reply:
> So for DSP_DMA_NONE we want to proceed to PlayDMATransfer

In the case of DSP_DMA_NONE, we still want to suppress dwords or less (as opposed to wide-open suppression which is what it previously was).

Will make the size_t change! Thanks, and for the background what's happening under the hood :-)

Reply:
Ok - it's all set.